kms/api-client/asymmetric_test.py

```diff
@@ -99,44 +100,54 @@ def test_get_public_key(self):
         assert isinstance(ec_key, _EllipticCurvePublicKey), 'expected EC key'
 
     def test_rsa_encrypt_decrypt(self):
-        ciphertext = sample.encryptRSA(self.message,
-                                       self.client,
-                                       self.rsaDecrypt)
+        ciphertext_bytes = sample.encryptRSA(self.message_bytes,
+                                             self.client,
+                                             self.rsaDecrypt)
+        ciphertext = base64.b64encode(ciphertext_bytes).decode()
         # ciphertext should be 344 characters with base64 and RSA 2048
         assert len(ciphertext) == 344, \
             'ciphertext should be 344 chars; got {}'.format(len(ciphertext))
-        assert ciphertext[-2:] == '==', 'cipher text should end with =='
-        plaintext = sample.decryptRSA(ciphertext, self.client, self.rsaDecrypt)
+        plaintext_bytes = sample.decryptRSA(ciphertext_bytes,
+                                            self.client,
+                                            self.rsaDecrypt)
+        assert plaintext_bytes == self.message_bytes
+        plaintext = plaintext_bytes.decode('utf-8')
```

Comment:
Is there much value in checking that plaintext == self.message if you've already checked that plaintext_bytes == self.message_bytes?

Reply:
Probably not, but I'd rather err on the side of too many tests, especially when dealing with all these encoding changes

Reply:
Most of the time you specify utf-8 for string.encode() and bytes.decode(), except on line 51 where you just assume the default (which is utf-8). Both ways work fine, but this is a minor inconsistency.

Reply:
good point, we should probably keep that consistent. Fixed